lib/game.rb

```diff
@@ -23,14 +23,30 @@ def make_move
     @display.display_board(@board.board)
     @display.show_current_player(@current_player.mark)
     move = @display.ask_for_move
-    @board.player_make_move(@current_player.mark, move)
-    @display.display_board(@board.board)
+    until move_valid?(move)
+      @display.invalid_move_message(move)
+      move = @display.ask_for_move
+    end
+    @board.player_make_move(@current_player.mark, move.to_i)
   end
 
   public
 
+  def moves_remaining?
+    @board.moves_remaining > 0
+  end
+
   def play_move
     make_move
     toggle_current_player
   end
+
+  def can_continue_playing?
```

Comment:
All this function does is calling another function. To me this overcomplicates things a bit.

Reply:
Think of this a "code smell"

Reply:
Thank you. This method now contains another method. They both consist of a condition to assess whether the game can continue